### ISSUE TITLE
Results Page Search and Return Buttons

### ISSUE THREAD

**Issue by ajburls** [dwyl/best-evidence#85]
On the results of a search, a search box appears at the top
![image](https://user-images.githubusercontent.com/30235568/28585023-d2594938-7166-11e7-8bb2-25be4df75a1e.png)
Not sure why, it doesn't work. Needs to work or be removed.

**Comment by SimonLab** [comment]
Thanks for spotting this @ajburls :+1: . I will update the input search to make it work on the results page.

**Comment by SimonLab** [comment]
This should be resolved now @ajburls

**Comment by ajburls** [comment]
@simonlab Just tested but, no, not searching on my android from search box in search answers screen - just returning me to the search page.

**Comment by SimonLab** [comment]
I've just try again and it's working on my phone. You might need to refresh the page to be sure to get the last version of the application.
I just want to check the user flow, we might have misunderstand the "done" link near the serach box. Do you expect this link to execute the new search? At the moment this link will indeed redirect to the search page so maybe it's something we need to clarify. Should we change this link to a "search" button and find another way to allow user to access the other pages of the applications (main search page, and soon search history page and notes page)

**Comment by ajburls** [comment]
I think I may have just found this box misleading. In my app it was blank and with the magnifying glass icon looked like a search box. It is now showing the search that was made.  I think it would be better to remove this icon as it has a specific meaning not relevant here. (Could put text like "Results for" or "search terms used") I think "done" is fine provided the search box is not misleading. (Note still getting double search record in list when I press done)

**Comment by SimonLab** [comment]
I'm adding some screenshot of the last version which will be soon deployed maybe it's going to clarify this feature. Let me know if this is the right user flow or if we should remove completely the search box on the results page

1 - Search for pain on the main search page:
![image](https://user-images.githubusercontent.com/6057298/28625692-ea47d146-7214-11e7-9b11-a69d95e8d4cc.png)

2 Click on the "search" button will display the result + add the term searched in the search box at the top of the page. The idea is for the user to specify the search term if needed without tapping again the term:
![image](https://user-images.githubusercontent.com/6057298/28625791-37fdf3d4-7215-11e7-913c-3ad228d85daa.png)

3 - We can do a new specify or do a new search from this page, here I search "water" **and press enter** this will update the results list:
![image](https://user-images.githubusercontent.com/6057298/28625911-7ca057c0-7215-11e7-9027-88c24b2963e2.png)

4 When I finish to search I can return to the main search page by clicking on "done" and I will see the term "water" and "pain" on my search history:
![image](https://user-images.githubusercontent.com/6057298/28625946-9b84c338-7215-11e7-909b-90cf10fdce17.png)
 
@ajburls is this user flow correct or should we update/delete some parts?

**Comment by ajburls** [comment]
@SimonLab Tested on desktop and android and looking good:
No duplicate searches appearing
"Done" returning me to search screen
Clicking on previous searches and text appearing in search box.

Only thing is I still thing the magnifying glass on search results page could mislead and would be better to be replaced by something like "Search terms:"

**Comment by markwilliamfirth** [comment]
@ajburls please can you test and close this issue?

**Comment by SimonLab** [comment]
We just need to remove the magnifying glass on mobile

**Comment by ajburls** [comment]
@SimonLab I though this was just giving the search terms used, but now I have found out that it is actually a search box, I think the magnifying glass is actually the most appropriate symbol (I just thought it was being used out of place).

Now I think the "Done" is in the wrong place given that this is a search box. I think it should say "search" and run the search.

We could have a Return button on left of box to go back to main page

**Comment by SimonLab** [comment]
I think changing the "done" link to a search button makes sense and we can keep the magnifying glass.
We only need to find a nice way to allow the user to go back to the main search page. A "return" button could work the only issue is to find the place for it on mobile.

**Comment by ajburls** [comment]
agreed

**Comment by markwilliamfirth** [comment]
Summary of action points:

- [ ] Change 'Done' link to a Search button for desktop and mobile

- [ ] Create a return button on left of the search terms box to go back to main page for desktop and mobile

Now that requirements have changed since the original time estimate @Cleop please could you time estimate these two tasks?

**Comment by markwilliamfirth** [comment]
@cleop please could you time estimate the above?

**Comment by markwilliamfirth** [comment]
Return button no longer required due to https://github.com/dwyl/best-evidence/issues/125#issuecomment-322701091 (home button)

**Comment by ajburls** [comment]
@SimonLab This is working now. Thanks.

I see that magnifying glass removed and button now a "search" button. (I am assuming that the bottom menu bar will be implemented soon allowing one to return to search screen directly, so absence of "done" button just a temporary hitch).

One thing: When I looked at this on my mobile the box for the search terms was quite small so only a couple of search terms were showing. It would be good if this box were wider. So, I think it would be better to just have magnifying glass rather than "search" button to allow search box to be bigger. I am not sure how big a task this is. I can close this issue if you want and open a new issue but it may be very quick to just make this alteration. Let me know.

**Comment by Cleop** [comment]
@ajburls - I've just made the change and will add the please-test label once it is merged in.

**Comment by ajburls** [comment]
@Cleop This looks good on the android phone (although it is quite hard to see magnifying glass perhaps a darker background for icon rather than grey would be better). Also, there is room for the search box to be wider and this would be better if this fairly straightforward to do.
![image](https://user-images.githubusercontent.com/30235568/29977336-f591f3aa-8f34-11e7-857b-1521ea8a47e2.png)

However, this is looking ugly on iPhone because magnifying glass has slipped into line above and this causes evidence type to appear as a white line in the red

![image](https://user-images.githubusercontent.com/30235568/29977566-bdff4798-8f35-11e7-935a-d5886bbd1803.png)

As a bit of an aside, I notice that the app looks like an app on the Android (no address bar) but appears as a webpage on the iPhone. Can it be made to look like an app on the iPhone?

**Comment by ajburls** [comment]
The search box could also be bigger on the desktop view.

The words search terms and search could be similarly replaced with the magnifying glass

![image](https://user-images.githubusercontent.com/30235568/29977739-58ad256c-8f36-11e7-99cc-234d26592ea2.png)

**Comment by ajburls** [comment]
@Cleop  the magnifying glasses now showing clearly with red background. And both on same line as search box. Thanks
